k2/csrc/ragged_ops_inl.h

```diff
@@ -624,6 +624,24 @@ void SegmentedExclusiveSum(Ragged<T> &src, Array1<T> *dst) {
   }
 }
 
+template <typename T>
+Ragged<T> CreateRagged2(const std::vector<std::vector<T>> &vecs) {
+  std::vector<T> values;
+  std::vector<int32_t> row_splits;
+  row_splits.reserve(vecs.size() + 1);
+  int32_t tot_size = 0;
+  row_splits.push_back(tot_size);
+  for (const auto &vec : vecs) {
+    tot_size += static_cast<int32_t>(vec.size());
+    row_splits.push_back(tot_size);
+    values.insert(std::end(values), std::begin(vec), std::end(vec));
```

Comment:
I would suggest `values.end()`, `vec.begin()`, and `vec.end()`, which requires
one fewer indirection, though your approach is fine.